ksql-cli/src/test/java/io/confluent/ksql/cli/SslFunctionalTest.java

```diff
@@ -0,0 +1,233 @@
+      .<String, String>builder()
+      .putAll(ClientTrustStore.trustStoreProps())
+      .build();
+
+  private static final EmbeddedSingleNodeKafkaCluster CLUSTER = EmbeddedSingleNodeKafkaCluster
+      .newBuilder()
+      .build();
+
+  private static final TestKsqlRestApp REST_APP = TestKsqlRestApp
+      .builder(CLUSTER::bootstrapServers)
+      .withProperties(ServerKeyStore.keyStoreProps())
+      .withProperty(RestConfig.LISTENERS_CONFIG, "https://localhost:0")
+      .build();
+
+  @ClassRule
+  public static final RuleChain CHAIN = RuleChain.outerRule(CLUSTER).around(REST_APP);
+
+  @Rule
+  public final ExpectedException expectedException = ExpectedException.none();
+
+  private Map<String, String> clientProps;
+
+  @Before
+  public void setUp() {
+    clientProps = Collections.emptyMap();
+  }
+
+  @Test
+  public void shouldNotBeAbleToUseCliWithIfClientDoesNotTrustServerCert() {
+    // Then:
+    expectedException.expect(KsqlRestClientException.class);
+    expectedException.expectCause(is(instanceOf(ProcessingException.class)));
+    expectedException.expectCause(hasCause(is(instanceOf(SSLHandshakeException.class))));
+
+    // When:
+    canMakeCliRequest();
+  }
+
+  @Test
+  public void shouldBeAbleToUseCliOverHttpsViaConfigFile() {
+    // Given:
+    givenTrustStoreConfigured();
+
+    // When:
+    final Code result = canMakeCliRequest();
+
+    // Then:
+    assertThat(result, is(Code.OK));
+  }
+
+  @Test
+  public void shouldBeAbleToUseCliOverHttpsViaSystemProps() {
+    // Given:
+    givenTrustStoreConfigured();
+
+    // When:
+    final Code result = canMakeCliRequest();
+
+    // Then:
+    assertThat(result, is(Code.OK));
+  }
+
+  @Test
+  public void shouldNotTableAbleToUseWssIfClientDoesNotTrustServerCert() throws Exception {
+    // Then:
+    expectedException.expect(SSLHandshakeException.class);
+    expectedException.expectCause(hasCause(hasMessage(containsString(
+        "unable to find valid certification path to requested target"))));
+
+    // When:
+    makeWsRequest(false);
+  }
+
+  @Test
+  public void shouldBeAbleToUseWss() throws Exception {
+    // When:
+    final Code result = makeWsRequest(true);
+
+    // Then:
+    assertThat(result, is(Code.OK));
+  }
+
+  private void givenTrustStoreConfigured() {
+    clientProps = TRUST_STORE_PROPS;
+  }
+
+  private Code canMakeCliRequest() {
+    final String serverAddress = REST_APP.getHttpsListener().toString();
+
+    try (KsqlRestClient restClient = new KsqlRestClient(serverAddress, emptyMap(), clientProps)) {
+
+      final RestResponse<?> response = restClient.makeKsqlRequest("show topics;");
+      if (response.isSuccessful()) {
+        return Code.OK;
+      }
+
+      return HttpStatus.getCode(response.getErrorMessage().getErrorCode());
+    }
+  }
+
+  private static Code makeWsRequest(final boolean setUpTrustStore) throws Exception {
+    final SslContextFactory sslContextFactory = new SslContextFactory();
+    if (setUpTrustStore) {
```

Comment:
nit: move this into "given" section and remove the flag since it's only used once anyway

Reply:
picky picky ;)